primitives/consensus/vrf/src/schnorrkel.rs

```diff
@@ -0,0 +1,208 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Schnorrkel-based VRF.
+
+use codec::{Encode, Decode};
+use sp_core::U512;
+use sp_runtime::RuntimeDebug;
+#[cfg(feature = "std")]
+use std::{ops::{Deref, DerefMut}, convert::TryFrom};
+#[cfg(feature = "std")]
+use codec::EncodeLike;
+#[cfg(feature = "std")]
+use schnorrkel::{SignatureError, errors::MultiSignatureStage};
+
+#[cfg(feature = "std")]
+pub use schnorrkel::vrf::{VRF_PROOF_LENGTH, VRF_OUTPUT_LENGTH};
+
+#[cfg(not(feature = "std"))]
+pub const VRF_PROOF_LENGTH: usize = 64;
+
+#[cfg(not(feature = "std"))]
+pub const VRF_OUTPUT_LENGTH: usize = 32;
+
+#[derive(Clone, Eq, PartialEq, RuntimeDebug, Encode, Decode)]
```

Comment:
let's add some docs to the remaining public types.

Reply:
Fixed!